be/src/exec/vectorized/jdbc_scanner.cpp

```diff
@@ -414,13 +416,14 @@ Status JDBCScanner::_fill_chunk(jobject jchunk, ChunkPtr* chunk) {
         jobject jcolumn = helper.list_get(jchunk, col_idx);
         DeferOp defer([&jcolumn, this]() { _jni_env->DeleteLocalRef(jcolumn); });
 
-#define FILL_COLUMN(get_value_func, cpp_type)                                                                         \
-    {                                                                                                                 \
-        auto func = std::bind(&JVMFunctionHelper::get_value_func, &helper, std::placeholders::_1);                    \
-        for (int i = 0; i < num_rows; i++) {                                                                          \
-            RETURN_IF_ERROR(                                                                                          \
-                    _append_value_from_result<cpp_type>(helper.list_get(jcolumn, i), func, slot_desc, column.get())); \
-        }                                                                                                             \
+#define FILL_COLUMN(get_value_func, cpp_type)                                                              \
+    {                                                                                                      \
+        auto func = std::bind(&JVMFunctionHelper::get_value_func, &helper, std::placeholders::_1);         \
+        for (int i = 0; i < num_rows; i++) {                                                               \
+            jobject jelement = helper.list_get(jcolumn, i);                                                \
+            DeferOp defer([&jelement, this]() { _jni_env->DeleteLocalRef(jelement); });                    \
+            RETURN_IF_ERROR(_append_value_from_result<cpp_type>(jelement, func, slot_desc, column.get())); \
```

Comment:
I think it's better to put this DeferOp inside _append_value_from_result, just like other _append_*_val functions

Reply:
done